pagopa/pn-frontend
# fix/PN-6358 analog flow timeline, change logic for title and a couple of descriptions

## Short description
This PR is about how to render the analog flow specific timeline events.  
Specifically, the **title** of the SEND_ANALOG_FEEDBACK events, which indicates whether the dispatch was successful or not, no longer depends on the value of `details.responseStatus` , but instead on `details.deliveryDetailCode`.  
Additionally, the **description** was changed for two values of `details.deliveryDetailCode`.

## List of changes proposed in this pull request
Changed `SendAnalogFlowStep` to change the way in which the title is computed. Specifically, whether the title is `details.timeline.send-analog-error` or `details.timeline.send-analog-success`.  
As indicated, this no longer depends on the value of `details.responseStatus` , but instead on `details.deliveryDetailCode`; for each possible value for SEND_ANALOG_FEEDBACK events, it is defined whether it corresponds to OK or KO outcome. This information is present in [this Confluence page](https://pagopa.atlassian.net/wiki/spaces/PN/pages/629014723/Timeline+events+e+cosa+mostrare+agli+utenti#Tabella-di-approfondimento-SEND_ANALOG).  
I defined two additional possible titles, one for the case in which the OK / KO outcome cannot be determined, other for the case in which the category is unknown. Neither of them could be obtained in no scenario given the current state of the codebase, they are included just for further eventual modifications.

Additionally, I changed (just in the `notifiche.json` files for PA / PF / PG) the descriptions for analog flow events with `deliveryDetailCode` RECRN004C or RECAG007C.

## How to test
For the changes in the title, just enter PA / PF / PG and open the detail of analog-flow notifications; maybe @federicofatica-pagopa or Chiara Tomei know something more.  
I indicate two examples present in the current state of TEST (I am not using DEV while load tests are in progress) 
- success: MAXV-WHRW-EGUL-202306-Z-1 / comune di Palermo / cristoforocolombo
- failure: ATJV-LRAZ-VGEH-202306-A-1 / comune di Palermo / cristoforocolombo

For the new copies, I don't know whether there are examples in DEV / TEST. I mocked the API changing the value of `deliveryDetailCode` for a timeline element verifying `category = SEND_ANALOG_PROGRESS`. An example for PA follows.

```
  getSentNotification: (iun: string): Promise<NotificationDetail> =>
    apiClient.get<NotificationDetail>(NOTIFICATION_DETAIL(iun)).then((response) => {
      if (response.data) {
        if (response.data.iun === 'MAXV-WHRW-EGUL-202306-Z-1') {
          const stepToModify = response.data.timeline.find(elem => elem.elementId === 'SEND_ANALOG_PROGRESS.IUN_MAXV-WHRW-EGUL-202306-Z-1.RECINDEX_0.ATTEMPT_0.IDX_1');
          if (stepToModify) {
            // eslint-disable-next-line functional/immutable-data
            (stepToModify.details as any).deliveryDetailCode = 'RECRN004C';
          }
        }        
        return parseNotificationDetail(response.data);
      }
      return {} as NotificationDetail;
    }),
```
